src/main/java/net/neoforged/neoforge/transfer/handlers/templates/contexts/StackItemContext.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.templates.contexts;
+
+import net.minecraft.world.SimpleContainer;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.neoforge.transfer.ResourceHandlerUtil;
+import net.neoforged.neoforge.transfer.handlers.IItemContext;
+import net.neoforged.neoforge.transfer.handlers.wrappers.items.VanillaContainerWrapper;
+import net.neoforged.neoforge.transfer.resources.ItemResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+
+public final class StackItemContext implements IItemContext {
```

Comment:
Assuming this isn't internal (at which point it should be annotated as such), this should have some docs on what this context does and what uses it has. As currently it makes it hard to review (especially for figuring out if `StackContainer` is implemented in a way that makes sense)